plugins/woocommerce/tests/php/src/Internal/DataStores/StockNotifications/StockNotificationsDataStoreTests.php

```diff
@@ -0,0 +1,535 @@
+	 */
+	private $data_store;
+
+	/**
+	 * Set up the test.
+	 */
+	public function setUp(): void {
+		parent::setUp();
+		$this->data_store = wc_get_container()->get( StockNotificationsDataStore::class );
+	}
+
+	/**
+	 * Tear down the test.
+	 */
+	public function tearDown(): void {
+		parent::tearDown();
+		// Clean up all notifications.
+		global $wpdb;
+		$wpdb->query( "TRUNCATE TABLE {$wpdb->prefix}wc_stock_notifications" );
+		$wpdb->query( "TRUNCATE TABLE {$wpdb->prefix}wc_stock_notificationmeta" );
+		$wpdb->query( "TRUNCATE TABLE {$wpdb->prefix}wc_stock_notifications_logs" );
+	}
+
+	/**
+	 * Test that the stock notification data store is registered.
+	 */
+	public function test_stock_notification_data_store_is_registered() {
+		$store = new \WC_Data_Store( 'stock_notification' );
+		$this->assertTrue( is_callable( array( $store, 'read' ) ) );
+		$this->assertEquals( StockNotificationsDataStore::class, $store->get_current_class_name() );
+	}
+
+	/**
+	 * Test creating a notification with all properties.
+	 */
+	public function test_create_notification_with_all_properties() {
+		$notification = new Notification();
+
+		// Set all properties.
+		$notification->set_product_id( 1 );
+		$notification->set_user_id( 1 );
+		$notification->set_user_email( 'test@test.com' );
+		$notification->set_status( 'active' );
+		$notification->set_date_created( '2024-01-01 00:00:00' );
+		$notification->set_date_modified( '2024-01-01 00:00:00' );
+		$notification->set_date_subscribed( '2024-01-01 00:00:00' );
+		$notification->set_date_notified( '2024-01-01 00:00:00' );
+		$notification->set_is_queued( true );
+
+		$notification->save();
+
+		// Verify all properties were saved correctly.
+		$this->assertEquals( 1, $notification->get_id() );
+		$this->assertEquals( 1, $notification->get_product_id() );
+		$this->assertEquals( 1, $notification->get_user_id() );
+		$this->assertEquals( 'test@test.com', $notification->get_user_email() );
+		$this->assertEquals( 'active', $notification->get_status() );
+		$this->assertEquals( '2024-01-01 00:00:00', $notification->get_date_created()->format( 'Y-m-d H:i:s' ) );
+		$this->assertEquals( '2024-01-01 00:00:00', $notification->get_date_modified()->format( 'Y-m-d H:i:s' ) );
+		$this->assertEquals( '2024-01-01 00:00:00', $notification->get_date_subscribed()->format( 'Y-m-d H:i:s' ) );
+		$this->assertEquals( '2024-01-01 00:00:00', $notification->get_date_notified()->format( 'Y-m-d H:i:s' ) );
+		$this->assertTrue( $notification->is_queued() );
+	}
+
+	/**
+	 * Test validation requirements for creating a notification.
+	 */
+	public function test_create_notification_validation() {
+		$notification = new Notification();
+
+		// Test missing product_id.
+		$result = $notification->save();
+		$this->assertInstanceOf( \WP_Error::class, $result );
+		$this->assertEquals( 'stock_notification_validation_error', $result->get_error_code() );
+		$this->assertEquals( 'Product ID is required', $result->get_error_message() );
```

Comment:
Not a biggie, but will this work with different language setting?

Reply:
Yup, I think it's a good idea to avoid using texts here. Updated in https://github.com/woocommerce/woocommerce/pull/57062/commits/e9087ae936272a870ec067e562415ac5defa5864